watcherx/directory.go

```diff
@@ -0,0 +1,120 @@
+package watcherx
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+)
+
+func listSubDirsDepth(parent string, depth int) (dirs []string, errs []error) {
+	if depth == 0 {
+		return
+	}
+	entries, err := ioutil.ReadDir(parent)
```

Comment:
Have you seen `filepath.Walk`?

https://yourbasic.org/golang/list-files-in-directory/

Reply:
Hm right for some reason I decided to not use it, I think because I misread the documentation and thought it would only list files and not directories.